python/lsst/resources/http.py

```diff
@@ -1137,6 +1218,116 @@ def walk(
                 new_uri = self.join(dir, forceDirectory=True)
                 yield from new_uri.walk(file_filter)
 
+    def generate_presigned_get_url(self, *, expiration_time_seconds: int) -> str:
+        """Return a pre-signed URL that can be used to retrieve this resource
+        using an HTTP GET without supplying any access credentials.
+
+        Parameters
+        ----------
+        expiration_time_seconds : `int`
+            Number of seconds until the generated URL is no longer valid.
+
+        Returns
+        -------
+        url : `str`
+            HTTP URL signed for GET.
+        """
+        return self._sign_with_macaroon(ActivityCaveat.DOWNLOAD, expiration_time_seconds)
+
+    def generate_presigned_put_url(self, *, expiration_time_seconds: int) -> str:
+        """Return a pre-signed URL that can be used to upload a file to this
+        path using an HTTP PUT without supplying any access credentials.
+
+        Parameters
+        ----------
+        expiration_time_seconds : `int`
+            Number of seconds until the generated URL is no longer valid.
+
+        Returns
+        -------
+        url : `str`
+            HTTP URL signed for PUT.
+        """
+        return self._sign_with_macaroon(ActivityCaveat.UPLOAD, expiration_time_seconds)
+
+    def to_fsspec(self) -> tuple[AbstractFileSystem, str]:
+        """Return an abstract file system and path that can be used by fsspec.
+
+        Returns
+        -------
+        fs : `fsspec.spec.AbstractFileSystem`
+            A file system object suitable for use with the returned path.
+        path : `str`
+            A path that can be opened by the file system object.
+        """
+        if self.isdir():
+            raise NotImplementedError(
+                f"method HttpResourcePath.to_fsspec() not implemented for directory {self}"
+            )
+
+        if fsspec is None:
+            raise ImportError("fsspec is not available")
+
+        async def get_client(**kwargs: Any) -> aiohttp.ClientSession:
+            # Create a SSL context equipped with the required certificates of
+            # the trusted certificate authorities. fsspect client must use
+            # that context to download the file.
+            ssl_context = ssl.create_default_context()
+            if self._config.ca_bundle is not None:
+                ssl_context.load_verify_locations(capath=self._config.ca_bundle)
+            return aiohttp.ClientSession(connector=aiohttp.TCPConnector(ssl=ssl_context), **kwargs)
+
+        # Retrieve a signed URL for a duration of 1 hour. We assume the URL
+        # is to be used only for downloading, to avoid the risk the caller
+        # modify its contents inadvertently.
+        url = self.generate_presigned_get_url(expiration_time_seconds=3_600)
```

Comment:
This will break `to_fsspec` for non-webdav HTTP/HTTPS URLs, since fetching a macaroon will fail.  Previously it was inheriting the base implementation from `ResourcePath.to_fsspec`, which just passes the URL unchanged to `fsspec.url_to_fs`.

Maybe add a conditional at the top delegating to the base class for non-webdav URLs?

Reply:
✅ Done: conditional added. `ResourcePath.to_fsspec` called for plain HTTP paths.